src/main/java/de/tum/in/www1/artemis/service/ProgrammingExerciseGradingService.java

```diff
@@ -215,6 +219,17 @@ public Result updateResult(Result result, ProgrammingExercise exercise, boolean
                 updatedResults.add(result);
             }
         }
+
+        // Update also manual results
+        List<StudentParticipation> participationsWithManualResult = participationService.findByExerciseIdWithManualResultAndFeedbacks(exercise.getId());
+        for (StudentParticipation studentParticipation : participationsWithManualResult) {
+            Result result = studentParticipation.findLatestResult();
+            if (result != null) {
+                updateResult(testCases, testCasesForCurrentDate, result, exercise);
```

Comment:
Maybe we need a bigger refactoring here. Instead of having an all in one `updateResult` method which gets more and more complex because it has to handle all kinds of different feedback, we create a `updateManualResult` method. This could also solve the problem that `updateResultString` triggers yet another calculation for the manual case. This might be out of scope of this PR.

Reply:
see general comment